src/charm.py

```diff
@@ -467,6 +526,15 @@ def _create_secret(self) -> Optional[Secret]:
         juju_secret = self.model.app.add_secret(secret, label=SECRET_LABEL)
         return juju_secret
 
+    def _migration_is_needed(self) -> Optional[bool]:
```

Comment:
Just have a minor question about why we return an `Optional`. We directly return `bool`. Unless the callers have different handling logic for `None` and `False`?

```
if not self._peers:
    return False
```

Reply:
The callers don't have different logic, because they check if the peer relation exists themselves. I wanted to be able distinguish between whether the migration has not run or if the peer relation is not ready, but this is not used right now. I could remove this check altogether.

Reply:
ic~ understood. I would like to say try not to use two falsy return values `None` and `False` together. The caller might be confused by those two return values. Maybe we could do the following:

```
def is_peer_relation_ready(self):
    return self._peers is not None

def is_migration_needed(self):
    return self._get_peer_data(PEER_KEY_DB_MIGRATE_VERSION) != self.kratos.get_version()

# In the caller function we do:
if self.is_peer_relation_ready() and self.is_migration_needed():
    do_something()
```